docs/recipes/UsageWithTypescript.md

```diff
@@ -6,259 +6,295 @@ hide_title: true
 
+## Typing Additional Redux Logic
+
 ### Type Checking Reducers
 
-Reducers are just pure functions that take the previous state, an action and then return the next state. In this example, we explicitly declare the type of actions this reducer will receive along with what it should return (the appropriate slice of state). With these additions TypeScript will give rich intellisense on the properties of our actions and state. In addition, we will also get errors when a certain case does not return the `ChatState`.
+[Reducers](../tutorials/fundamentals/part-3-state-actions-reducers.md) are pure functions that receive the current `state` and incoming `action` as arguments, and return a new state.
 
-Type checked chat reducer:
+If you are using Redux Toolkit's `createSlice`, you should rarely need to specifically type a reducer separately. If you do actually write a standalone reducer, it's typically sufficient to declare the type of the `initialState` value, and type the `action` as `AnyAction<string>`:
 
 ```ts
-// src/store/chat/reducers.ts
+import { AnyAction } from 'redux'
 
-import {
-  ChatState,
-  ChatActionTypes,
-  SEND_MESSAGE,
-  DELETE_MESSAGE
-} from './types'
+interface CounterState {
+  value: number
+}
 
-const initialState: ChatState = {
-  messages: []
+const initialState: CounterState = {
+  value: 0
 }
 
-export function chatReducer(
+export default function counterReducer(
   state = initialState,
-  action: ChatActionTypes
-): ChatState {
-  switch (action.type) {
-    case SEND_MESSAGE:
-      return {
-        messages: [...state.messages, action.payload]
-      }
-    case DELETE_MESSAGE:
-      return {
-        messages: state.messages.filter(
-          message => message.timestamp !== action.meta.timestamp
-        )
-      }
-    default:
-      return state
-  }
+  action: AnyAction<string>
+) {
+  // logic here
 }
 ```
 
-Type checked system reducer:
+However, the Redux core does export a `Reducer<State, Action>` type you can use as well.
 
-```ts
-// src/store/system/reducers.ts
+### Type Checking Middleware
 
-import { SystemState, SystemActionTypes, UPDATE_SESSION } from './types'
+[Middleware](../tutorials/fundamentals/part-4-store.md#middleware) are an extension mechanism for the Redux store. Middleware are composed into a pipeline that wrap the store's `dispatch` method, and have access to the store's `dispatch` and `getState` methods.
 
-const initialState: SystemState = {
-  loggedIn: false,
-  session: '',
-  userName: ''
-}
+The Redux core exports a `Middleware` type that can be used to correctly type a middleware function:
 
-export function systemReducer(
-  state = initialState,
-  action: SystemActionTypes
-): SystemState {
-  switch (action.type) {
-    case UPDATE_SESSION: {
-      return {
-        ...state,
-        ...action.payload
-      }
-    }
-    default:
-      return state
-  }
-}
+```ts
+export interface Middleware<
+  DispatchExt = {}, // legacy type parameter that is no longer relevant
```

Comment:
`DispatchExt` is what we use extensively in redux toolkit.
It is used to add additional function overloads ("extensions") to the resulting `store.dispatch`